### PR TITLE
fix: #242 ensuring alias is set or raise error.

### PR DESCRIPTION
**Pull request by javydekoning**
*Issue #, if available:*
#242

*Description of changes:*
- Ensuring an error is raised if the account alias is non-unique. 
- Update the `readme.md` to reflect that.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Test execution (SHOULD error):

```
2022-04-03 13:17:47,547 \| ERROR \| configure_account_alias \| The account alias security already exists. The account alias must be unique across all Amazon Web Services products. Refer to https://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html#AboutAccountAlias \| (configure_account_alias.py:26)
--
[ERROR] EntityAlreadyExistsException: An error occurred (EntityAlreadyExists) when calling the CreateAccountAlias operation: The account alias security already exists.Traceback (most recent call last):  File "/var/task/configure_account_alias.py", line 42, in lambda_handler    create_account_alias(event, role.client("iam"))  File "/var/task/configure_account_alias.py", line 30, in create_account_alias    raise error  File "/var/task/configure_account_alias.py", line 24, in create_account_alias    iam_client.create_account_alias(AccountAlias=account.get("alias"))  File "/var/task/botocore/client.py", line 401, in _api_call    return self._make_api_call(operation_name, kwargs)  File "/var/task/aws_xray_sdk/ext/botocore/patch.py", line 38, in _xray_traced_botocore    return xray_recorder.record_subsegment(  File "/var/task/aws_xray_sdk/core/recorder.py", line 435, in record_subsegment    return_value = wrapped(*args, **kwargs)  File "/var/task/botocore/client.py", line 731, in _make_api_call    raise error_class(parsed_response, operation_name)
```

### CODE REVIEW

**Review by StewartW — DISMISSED** [review]

**Comment by javydekoning** [comment]
@StewartW yes, I've added a test. Thanks for reviewing!

**Review by StewartW — DISMISSED** [review]

File: src/lambda_codebase/account_processing/configure_account_alias.py [hunk 1]
```diff
@@ -22,8 +22,13 @@ def create_account_alias(account, iam_client):
...
     try:
         iam_client.create_account_alias(AccountAlias=account.get("alias"))
-    except iam_client.exceptions.EntityAlreadyExistsException:
-        pass
+    except iam_client.exceptions.EntityAlreadyExistsException as error:
+        LOGGER.error(
+            """The account alias security already exists.
+            The account alias must be unique across all Amazon Web Services products.
+            Refer to https://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html#AboutAccountAlias"""
+        )
```
Comment (sbkok) [inline src/lambda_codebase/account_processing/configure_account_alias.py:30]: Could you ensure this shows the alias that is in use?
At the moment it refers to the `security` alias :)

Plus could you convert this to the multi-line strings that we use elsewhere, like this?

```suggestion
        LOGGER.error(
            f"The account alias ${account.get('alias')} already exists."
            "The account alias must be unique across all Amazon Web Services products."
            "Refer to https://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html#AboutAccountAlias"
        )
```

File: src/lambda_codebase/account_processing/tests/test_account_alias.py [hunk 2]
```diff
@@ -2,24 +2,47 @@
...
+            'EntityAlreadyExistsException',
+            'An error occurred (EntityAlreadyExists) when calling the CreateAccountAlias operation: The account alias nonunique already exists.'
+        )
+        stubber.activate()
+
+        with self.assertRaises(ClientError) as _error:
+            create_account_alias(test_account, iam_client)
+        self.assertRegex(
+            str(_error.exception),
+            r'.*The account alias nonunique already exists.*'
```
Comment (sbkok) [inline src/lambda_codebase/account_processing/tests/test_account_alias.py:47]: I'm afraid this test does not work at the moment.
Can you verify this works after applying the suggestion above?

**Review by sbkok — CHANGES_REQUESTED** [review]

File: src/lambda_codebase/account_processing/configure_account_alias.py (hunk 1 above)
Comment (javydekoning) [reply src/lambda_codebase/account_processing/configure_account_alias.py]: Thanks for spotting @sbkok , I've committed a fix.

File: src/lambda_codebase/account_processing/tests/test_account_alias.py (hunk 2 above)
Comment (javydekoning) [reply src/lambda_codebase/account_processing/tests/test_account_alias.py:47]: Test works now! Thanks!

**Review by sbkok — APPROVED** [review]
LGTM, thank you!

**Review by StewartW — APPROVED** [review]

**Pull request merged**